ganga/GangaCore/GPIDev/Lib/GangaList/GangaList.py

```diff
@@ -670,6 +670,46 @@ def accept(self, visitor):
 
         visitor.nodeEnd(self)
 
+    # TODO: Remove the @synchronised, once locks are removed altogether
+    @synchronised
```

Comment:
If we want to keep both the XML and the database implementations working at the same time, we need a discussion about what to do with locking. Can we just change what @synchronised does depending on persistency is used (and maybe change its name to avoid future confusion).